lightning/src/chain/channelmonitor.rs

```diff
@@ -2351,13 +2685,39 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 								// resolve the source HTLC with the original sender.
 								payment_data = Some(((*source).clone(), htlc_output.payment_hash, htlc_output.amount_msat));
 							} else if !$holder_tx {
-									check_htlc_valid_counterparty!(self.current_counterparty_commitment_txid, htlc_output);
+								check_htlc_valid_counterparty!(self.current_counterparty_commitment_txid, htlc_output);
 								if payment_data.is_none() {
 									check_htlc_valid_counterparty!(self.prev_counterparty_commitment_txid, htlc_output);
 								}
 							}
 							if payment_data.is_none() {
 								log_claim!($tx_info, $holder_tx, htlc_output, false);
+								let outbound_htlc = $holder_tx == htlc_output.offered;
+								if !outbound_htlc || revocation_sig_claim {
+									self.onchain_events_awaiting_threshold_conf.push(OnchainEventEntry {
+										txid: tx.txid(), height,
+										event: OnchainEvent::HTLCSpendConfirmation {
+											input_idx: input.previous_output.vout,
+											preimage: if accepted_preimage_claim || offered_preimage_claim {
+												Some(payment_preimage) } else { None },
+											// `payment_data.is_none()` implies that this is a payment
+											// to us, as we haven't learned anything to cause us to
+											// update another channel or our offchain state. Thus, wait
+											// for the CSV delay before dropping the HTLC from
+											// claimable balance if the claim was an HTLC-Success or
+											// HTLC-Timeout transaction.
+											on_to_local_output_csv: if accepted_preimage_claim || offered_timeout_claim {
+												Some(self.on_holder_tx_csv) } else { Some(0) },
```

Comment:
If the HTLC is inbound (`!outbound_htlc`), it means it's either a received HTLC ouput on holder commitment or an offered HTLC output on counterparty commitment. If it's an offered HTLC output on counterparty commitment, it can be claimed with an `offered_preimage_claim` and _not_ a `offered_timeout_claim` ? And further, if it's a `offered_preimage_claim`, transaction isn't encumbered by a CSV ?

Reply:
This is inside of an if block that checks `!outbound_htlc || revocation_sig_claim`, so it can't be an inbound HTLC that is claimed by anything other than a revoked claim (which we don't handle anyway).